r/tests/testthat/test-Array.R

```diff
@@ -749,3 +749,17 @@ test_that("Array$ApproxEquals", {
   expect_true(a$ApproxEquals(b))
   expect_false(a$ApproxEquals(vec))
 })
+
+test_that("auto int64 conversion to int can be disabled (ARROW-10093)", {
+  op <- options(arrow_disable_int64_auto_conversion = TRUE); on.exit(options(op))
```

Comment:
Re: naming, we currently have options "arrow.use_threads" (mostly private but some may need to set it given the multithreading bugs we've observed) and "arrow.dev.repo" (not advertised). I don't feel strongly about the naming convention as long as we're consistent and predictable (which, naturally, `names(options())` is not). 

Searching around for other conventions, I found `package.option_name` (cf. https://github.com/tidyverse/dplyr/pull/5548) like we did with `arrow.use_threads`, so maybe let's standardize on that?

Reply:
using `arrow.int64_downcast`, happy to change